src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/AADFedAuthTokenRefreshTest/AADFedAuthTokenRefreshTest.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class AADFedAuthTokenRefreshTest
+    {
+        private readonly ITestOutputHelper _testOutputHelper;
+
+        public AADFedAuthTokenRefreshTest(ITestOutputHelper testOutputHelper)
+        {
+            _testOutputHelper = testOutputHelper;
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsAADPasswordConnStrSetup))]
+        public void FedAuthTokenRefreshTest()
+        {
+            string connStr = DataTestUtility.AADPasswordConnectionString;
+
+            // Create a new connection object and open it
+            using (SqlConnection connection = new SqlConnection(connStr))
+            {
+                connection.Open();
+
+                // Set the token expiry to expire in 1 minute from now to force token refresh
+                string tokenHash1 = "";
+                DateTime? oldExpiry = GetOrSetTokenExpiryDateTime(connection, true, out tokenHash1);
+                Assert.True(oldExpiry != null, "Failed to make token expiry to expire in one minute.");
+
+                // Convert and display the old expiry into local time which should be in 1 minute from now
+                DateTime oldLocalExpiryTime = TimeZoneInfo.ConvertTimeFromUtc((DateTime)oldExpiry, TimeZoneInfo.Local);
+                LogInfo($"Token: {tokenHash1}   Old Expiry: {oldLocalExpiryTime}");
+                TimeSpan timeDiff = oldLocalExpiryTime - DateTime.Now;
+                Assert.True(timeDiff.TotalSeconds <= 60, "Failed to set expiry after 1 minute from current time.");
```

Comment:
```suggestion
                Assert.InRange(timeDiff.TotalSeconds, 0, 60);
```

Reply:
The Assert.InRange is now in used as suggested.